src/libstd/sys/windows/fs.rs

```diff
@@ -575,3 +575,12 @@ pub fn canonicalize(p: &Path) -> io::Result<PathBuf> {
         PathBuf::from(OsString::from_wide(buf))
     })
 }
+
+pub fn copy(from: &Path, to: &Path) -> io::Result<u64> {
+    let pfrom = to_utf16(from);
+    let pto = to_utf16(to);
+    try!(cvt(unsafe {
+        c::CopyFileW(pfrom.as_ptr(), pto.as_ptr(), libc::FALSE)
+    }));
+    stat(to).map(|attr| attr.size())
```

Comment:
I'm a little worried about this implementation because it means that the file could be successfully copied but then this later call to `stat` could fail, causing the entire operation to be considered as failing. Basically the atomicity of this operation means that it can be somewhat significantly different from the unix implementation.